auditbeat/_meta/fields.common.yml

```diff
@@ -77,10 +83,16 @@
       - name: id
         type: keyword
         description: Saved user ID.
+      - name: name
+        type: keyword
+        description: Saved user name.
       - name: group
         type: group
         description: Saved group information.
         fields:
         - name: id
           type: keyword
           description: Saved group ID.
+        - name: name
+          type: keyword
+          description: Saved group name.
```

Comment:
To clarify about FIM above: I would define the fields in this PR. Addressing whether FIM emits owner IDs in these fields can be addressed as a separate PR.

Reply:
> I find it confusing to have the `effective` & `saved` fields defined here, and the `audit` & `filesystem` fields defined only in the auditd module. Especially since they're defined mixed in the ECS `user` field set.
> 
> I think it would make most sense to have them grouped together here.

+1 I'll make the change

> I would assume Auditbeat's FIM has events with `ouid` and `ogid` (file owner). I think it would make sense to define them here as (and adjust FIM to use them ;-) ).

I think we have a few options. But regardless, I don't think we should address it here in this PR.

The FIM [file metadata](https://github.com/elastic/beats/blob/master/auditbeat/module/file_integrity/event.go#L126) has a `uid`, `gid`, `owner`, and `group`. This is also what is [currently in ECS](https://github.com/elastic/ecs#-file-fields).

Our options for moving the fields:

1. Move them under `user.*`: `user.id`, `user.name`, `user.group.id`, `user.group.name` - no new definitions required, easy field names, but problematic if we ever have another user object (e.g. the user that modified the file)
2. Move them under `user.owner.*`: `user.owner.id`, `user.owner.name`, `user.owner.group.id`, `user.owner.group.name` - more future proof, somewhat longer field names
3. Move them under `file.user.*`: `file.user.owner.id`, `file.user.owner.name`, `file.user.owner.group.id`, `file.user.owner.group.name`
4. Move them under `file.owner.*`: `file.owner.id`, `file.owner.name`, `file.owner.group.id`, `file.owner.group.name`

The benefit the last two options under `file` have is that it would allow us to keep the `file` object of the FIM module and the `auditd.paths` object (containing multiple files) of the Auditd module to share the same fields for users. They're a bit disparate at the moment, with FIM using `uid/gid` and Auditd using `ouid/ogid` for the same data.

Again, I think we should address this whole thing in another PR, including any field definitions.

Reply:
Thanks for moving the fields defs :-)

I agree the owner stuff is to be addressed in another PR, if we address it.

The discrepancy in representing users in `file.*` in ECS saddens me a bit, haha. This one slipped by. But I think we should leave it as is and concentrate on polishing everything that's already in flight, for FF. I don't think we should open this can of worms as well.